pkg/extension/wasmimage/pusher.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2021 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wasmimage
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+
+	"github.com/containerd/containerd/remotes"
+	"github.com/deislabs/oras/pkg/auth/docker"
+	orasctx "github.com/deislabs/oras/pkg/context"
+	"github.com/deislabs/oras/pkg/oras"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+var (
+	pushOpts = []oras.PushOpt{
+		oras.WithConfigMediaType(ConfigMediaType),
+		oras.WithNameValidation(nil),
+	}
+)
+
+// PusherOpts represents options for Pusher
+type PusherOpts struct {
+	AllowInsecure bool
+	UseHTTP       bool
+}
+
+// NewPusherOpts returns a default PusherOpts instance
+func NewPusherOpts() PusherOpts {
+	return PusherOpts{
+		AllowInsecure: false,
+		UseHTTP:       false,
+	}
+}
```

Comment:
I don't think this struct is necessary. I suggest just parse the parse flags with local variables and then pass them to `NewPusher` directly.

Reply:
or put them in `cmdOpts` just like others.